extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+     * @private {?Object}
+     * */
+    this.options_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** Populating parameters once */
+    this.widgetId_ = this.element.getAttribute('data-id');
+    this.url_ = this.element.getAttribute('data-url');
+    this.key_ = this.element.getAttribute('data-key');
+    this.options_ = this.parseOptions_();
+    let domain = 'cdn.iframe.ly';
+    const requestedDomain = this.element.getAttribute('data-domain');
+    if (requestedDomain && this.isValidDomain_(requestedDomain)) {
+      domain = requestedDomain;
+    }
+    this.base_ = `https://${domain}/`;
+    this.parseAttributes_();
+    setStyle(this.element, 'box-sizing', 'border-box');
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    if (this.preconnect && this.preconnect.url_) {
+      this.preconnect.url(this.base_, opt_onLayout);
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    /**
+     * Building placeholder for responsive layout without resizing
+     * or when requested via `data-img` attribute
+     */
+    const layout = this.getLayout();
+    const resizable = this.element.getAttribute('resizable') !== null;
+    const isImg = this.element.getAttribute('data-img') !== null;
+
+    if (isImg || (layout === 'responsive' && !resizable)) {
+      /** using Iframely placeholder image */
+      let src = this.constructPlaceholderSrc_();
+      src = addParamsToUrl(src, this.options_);
+      return createElementWithAttributes(
+        this.element.ownerDocument,
+        'amp-img',
+        {
+          'src': src,
+          'layout': 'fill',
+          'placeholder': '',
+        }
+      );
+    } else {
+      return null;
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    /** attach iFrame */
+    const me = this;
+    this.iframe_ = this.element.ownerDocument.createElement('iframe');
+    setStyle(this.iframe_, 'border', '0');
+    this.iframe_.setAttribute(
+      'allow',
+      'encrypted-media *; accelerometer *; gyroscope *; picture-in-picture *; camera *; microphone *;'
+    );
+    this.iframe_.setAttribute(
+      'sandbox',
+      'allow-scripts allow-same-origin allow-popups allow-forms allow-presentation'
+    );
+    this.src_ = addParamsToUrl(this.src_, this.options_);
+    this.iframe_.src = this.src_;
+    this.applyFillContent(this.iframe_);
+    this.element.appendChild(this.iframe_);
+
+    /**
+     * Handling Iframely messages
+     * @param {window.event.source} event - event to check
+     * */
+    function receiveMessage(event) {
+      const iframes = me.element.getElementsByTagName('iframe');
+      if (findIframeByContentWindow(iframes, event.source)) {
+        me.handleEvent(me, event);
+      }
+    }
+    window.addEventListener('message', receiveMessage, false);
```

Comment:
1. Use `event-helper.js`#`listen()`, `this.win`.
2. You're already keeping the `iframe_`, why the lookup?
3. Keep resulting unlistener for unlayout.
4. Private `handleEvent_`.
5. [Do you know arrow functions?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) You don't need that `me` thing.

```suggestion
    this.unlistener_ = listen(this.win, 'message', (event) => {
      if (event.source === this.iframe_.contentWindow) {
        this.handleEvent_(event);
      }
    });
```